### PR TITLE
make: Use makefile wildcard function instead of shell ls

### PR DESCRIPTION
**Pull request by gebart**
I had forgot to PR this branch before..
Alternative to #4424

### CODE REVIEW

**Comment by cgundogan** [comment]
I am wondering if it's really necessary to put each Makefile change into a single commit? I probably would reduce this to 3 commits: one for native, one containing all boards, and one for desvirt.
The change to the board Makefiles is straight forward and if it breaks for one it should break for all. Reverting a single commit is simpler than reverting them all, and the commit history doesn't look that bloated. But that's just a matter of taste (:

**Comment by gebart** [comment]
@cgundogan OK, let's squash it after review. Can someone test this on OSX to see if it fixes the problem described in #4424 (`no such file or directory /dev/blah`)?

**Comment by cgundogan** [comment]
@thomaseichinger can you have a look at this?

**Comment by thomaseichinger** [comment]
I'm in favore of this PR, it reduces build time by 300ms on my system and I prefer staying in one scripting language without invoking another interpreter. Will close #4424, @gebart please squash.

**Comment by cgundogan** [comment]
@thomaseichinger was that a silent ACK? (;

**Comment by thomaseichinger** [comment]
@cgundogan No because I still have to test it but that's independent from the squashing. And as we have H&A upcoming I don't want to hesitate and upset Travis already before starting. :wink:

**Comment by gebart** [comment]
squashed

File: boards/native/Makefile.include [hunk 1]
```diff
@@ -146,9 +146,9 @@ term-cachegrind:
 term-gprof: term
 
 eval-gprof:
-	$(GPROF) $(ELF) $(shell ls -rt gmon.out* | tail -1)
+	$(GPROF) $(ELF) $(shell ls -t gmon.out* | head -1)
 
 eval-cachegrind:
-	$(CGANNOTATE) $(shell ls -rt cachegrind.out* | tail -1)
+	$(CGANNOTATE) $(shell ls -t cachegrind.out* | head -1)
```
Comment (LudwigKnuepfer) [inline boards/native/Makefile.include]: This seems to be wrong.

Comment (LudwigKnuepfer) [inline boards/native/Makefile.include]: IE: The commit message is off.

**Comment by kaspar030** [comment]
needs rebase.

**Comment by gebart** [comment]
rebased, removed native nonsense commit, removed desvirt commit since the shell call was refactored out by some other PR.

**Comment by Kijewski** [comment]
ACK

**Comment by gebart** [comment]
Restarted Travis

**Pull request merged**